### PR TITLE
Separate out client and server clusters in IDL

### PR DESCRIPTION
**Pull request by andy31415**
#### Problem
Zigbee data model has client and server clusters.

This does not make sense in matter as client clusters could be on any endpoint. 

#### Change overview
List server  clusters on endpoints, all other clusters are marked  as 'uses' to help review in what is being code-generated.

#### Testing
IDL generated.

### CODE REVIEW

**Comment by bzbarsky-apple** [comment]
> This does not make sense in matter as client clusters could be on any endpoint.

Not true if bindings are going to work.  Client clusters on devices like light switches need to be on a specific endoint that you can then bind to things.

Client clusters on things like phone apps or whatnot, that don't implement any server bits at all, can do whatever, of course.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]
The endpoint a client cluster is on matters.

File: src/app/zap-templates/templates/chip/helper.js [hunk 1]
```diff
@@ -453,6 +453,11 @@ async function if_chip_enum(type, options)
   return templateUtil.templatePromise(this.global, result);
 }
 
+function if_server_side(options)
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/templates/chip/helper.js:456]: Oh and the normal idiom here is:
```
{#if (isServer side)}}
```

**Comment by andy31415** [comment]
Had chat with @bzbarsky-apple - it seems that the data model that zap exposes for clusters is representing 'bindings'.

I will close the PR and add side in some form inside the endpoint.

**Pull request closed without merging**